.github/workflows/publish-package.yml

```diff
@@ -36,11 +45,25 @@ jobs:
       - name: Build
         run: yarn build
 
-      # TODO: we still need to make the tests work in CI environments
-      # will come back to this
-      # - name: Test
-      #   run: yarn test
+      - name: E2E tests
+        id: e2e
+        env:
+          API_KEY: ${{ secrets.API_KEY }}
+          OWNER_MNEMONIC: ${{ secrets.OWNER_MNEMONIC }}
+          PAYMASTER_POLICY_ID: ${{ secrets.PAYMASTER_POLICY_ID }}
+        run: yarn test:e2e
+
+      - name: Artifact of E2E tests
+        id: artifact
+        if: steps.e2e.outcome == 'success'
```

Comment:
from [here](https://stackoverflow.com/a/70549615) on how to get the result of the e2e step. referenced by id.

Reply:
is this necessary, I thought this is what `needs` is for below? like if one of these things fails causing `build_and_test` to fail, then the needs constraint isn't met? I could be wrong actually because it's been a minute since I've used needs

Reply:
https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-requiring-successful-dependent-jobs yea I think you should be fine with just needs

Reply:
i just realized i can just do the conditionals on a job-step, not just jobs. so just going to consolidate deploy into the build_and_test job to become build_and_test_and_deploy, conditional on the publish flag and successful e2e. 

this avoids duplicating checkout, install, and build (per your below comment)